lib/elixir/src/elixir_expand.erl

```diff
@@ -380,7 +380,8 @@ expand({Name, Meta, Kind}, S, E) when is_atom(Name), is_atom(Kind) ->
           {{Name, Meta, Kind}, S, E};
 
         _ ->
-          function_error(Meta, E, ?MODULE, {undefined_var, Name, Kind}),
+          SpanMeta = [calculate_span(Name, Meta) | Meta],
+          function_error(SpanMeta, E, ?MODULE, {undefined_var, Name, Kind}),
           {{Name, Meta, Kind}, S, E}
```

Comment:
```suggestion
          function_error(SpanMeta, E, ?MODULE, {undefined_var, Name, Kind})
```

Right?

Reply:
You meant passing `SpanMeta` instead of `Meta` in the returning tuple or really remove it?

Reply:
I think we don t need the `{{Name, Meta, Kind}, S, E}` because function_error always raises?

Reply:
does it? I've tried removing it and some tests broke, I think we don't raise here because this is the code path the diagnostics go through, we just print them to stderr and exit gracefully AFAIK

Reply:
Ah, i forgot we now collect multiple diagnostics. Sorry :)